CMakeLists.txt

```
@@ -433,13 +433,19 @@ set(IREE_CUDA_LIBDEVICE_PATH "" CACHE STRING "Absolute path to an appropriate li
 if(IREE_TARGET_BACKEND_CUDA OR IREE_HAL_DRIVER_CUDA)
   find_package(CUDAToolkit)
 
-  # We define the magic IREE_CUDA_DEPS_DIR env var in our CI docker images if we
-  # have a stripped down CUDA toolkit suitable for compiling available. We
-  # trigger on this below as a fallback for locating headers and libdevice
-  # files.
   if(NOT CUDAToolkit_FOUND AND DEFINED ENV{IREE_CUDA_DEPS_DIR})
+    # We define the magic IREE_CUDA_DEPS_DIR env var in our CI docker images if we
+    # have a stripped down CUDA toolkit suitable for compiling available. We
+    # trigger on this below as a fallback for locating headers and libdevice
+    # files.
     set(CUDAToolkit_ROOT "$ENV{IREE_CUDA_DEPS_DIR}")
     message(STATUS "CUDA SDK not found by CMake but using IREE_CUDA_DEPS = ${CUDAToolkit_ROOT}")
+  elseif(NOT CUDAToolkit_FOUND AND NOT DEFINED ENV{IREE_CUDA_DEPS_DIR})
+    # If we haven't found CUDA deps, download at least enough to build for CUDA.
+    # This will define IREE_CUDA_DOWNLOAD_LIBDEVICE_PATH & IREE_CUDA_DOWNLOAD_INCLUDE_PATH
+    # vars with the target deps.
+    message(STATUS "CUDA SDK not found by CMake but downloading dependencies")
+    add_subdirectory(build_tools/third_party/cuda EXCLUDE_FROM_ALL)
   endif()
```

I think this would be more readable if it was broken into more nesting (which is probably why we missed the error previously).


```suggestion
  if(NOT CUDAToolkit_FOUND)
    if(DEFINED ENV{IREE_CUDA_DEPS_DIR})
      # We define the magic IREE_CUDA_DEPS_DIR env var in our CI docker images if we
      # have a stripped down CUDA toolkit suitable for compiling available. We
      # trigger on this below as a fallback for locating headers and libdevice
      # files.
      set(CUDAToolkit_ROOT "$ENV{IREE_CUDA_DEPS_DIR}")
      message(STATUS "CUDA SDK not found by CMake but using IREE_CUDA_DEPS = ${CUDAToolkit_ROOT}")
    else()
      # If we haven't found CUDA deps, download at least enough to build for CUDA.
      # This will define IREE_CUDA_DOWNLOAD_LIBDEVICE_PATH & IREE_CUDA_DOWNLOAD_INCLUDE_PATH
      # vars with the target deps.
      message(STATUS "CUDA SDK not found by CMake but downloading dependencies")
      add_subdirectory(build_tools/third_party/cuda EXCLUDE_FROM_ALL)
    endif()
  endif()
```